src/gset.rs

```diff
@@ -79,4 +79,23 @@ impl<T: Ord> GSet<T> {
     pub fn contains(&self, element: &T) -> bool {
         self.value.contains(element)
     }
+
+    /// Returns the `BTreeSet` for this `GSet`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use crdts::GSet;
+    /// use std::collections::BTreeSet;
+    /// let mut a = GSet::new();
+    /// let mut b = BTreeSet::new();
+    /// for i in 1..10 {
+    ///     a.insert(i);
+    ///     b.insert(i);
+    /// }
+    ///
+    /// assert_eq!(a.read(), b);
+    pub fn read(self) -> BTreeSet<T> {
```

Comment:
To follow the API of the rest of the CRDT's in the library, lets not consume `self` here and just return a clone.

ie.
``` rust
pub fn read(&self) -> BTreeSet<T> {
    self.value.clone()
}
```

And then also provide an `impl From<GSet<T>> for BTreeSet<T>` or an `impl Into<BTreeSet<T>> for GSet<T>` for people who don't want to allocate memory.
https://doc.rust-lang.org/std/convert/trait.Into.html

Reply:
Ya fair enough. I came across this a while back https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control, which recommends the opposite. Just food for thought :)

Reply:
Ah, but I suppose we provide that by adding the `From` :+1: